### ISSUE TITLE
Application Upgrade error BC15 CU1

### ISSUE THREAD

**Issue by krregg** [MicrosoftDocs/dynamics365smb-devitpro-pb#876]
Hi,

I tried to upgrade BC14 CU6 to BC15 CU1 but I encountered problems during an application upgrade. 

I am following this MS post: https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/upgrade/upgrade-unmodified-application

And I get stuck on step 7 when I run the command:
`Start-NAVDataUpgrade -ServerInstance BC150 -FunctionExecutionMode Serial -SkipAppVersionCheck`

When I check the progress using `Get-NAVDataUpgrade` I can see the progress bar is stopped on 74,07 %. I checked also Event Viewer where I can see this:

`Message Application upgrade error codeunit id = 2, method = OnRun, company = CRONUS International Ltd., exception = Microsoft.Dynamics.Nav.Types.Exceptions.NavNCLMetadataObjectNotFoundException: Object of type CodeUnit with ID 2 could not be found.`

I am working with the default Cronus database, no application modification but the upgrade doesn't work as expected regarding MS instructions. 

What am I missing in this process? I tried a few times, the same result...

**Comment by jswymer** [comment]
Hello, Are you running a single tenant server instance? If so, and it is stuck at a percentage under 100, try to restart the server instance. Then, run Get-NAVAppInfo -ServerInstance BC150 -Tenant Default and see if the Base application and System applications are installed.

**Comment by krregg** [comment]
Hi @jswymer , thank you for your reply.

Yes, I am running single-tenant, locally on my computer.

I did as you mentioned and I can see that Base and System app-s are installed.

```Id            : 437dbf0e-84ff-417a-965d-ed2bb9650972
Name          : Base Application
Version       : 15.1.37881.38071
Publisher     : Microsoft
ExtensionType : ModernDev
Scope         : Global

Id            : 63ca2fa4-4f03-4f2b-a480-172fef340d3f
Name          : System Application
Version       : 15.1.37881.38071
Publisher     : Microsoft
ExtensionType : ModernDev
Scope         : Global```

I also can reach http://localhost:8080/BC150/.

After I restarted the service and I run `Start-NAVDataUpgrade -ServerInstance BC150 -Tenant default -FunctionExecutionMode Serial -SkipAppVersionCheck` which seems to be ok now, it finished. Also if I check `Get-NAVDataUpgrade` I can't see any loading-bar anymore. I guess this is it?

So if I understood right, I just have to restart the service when performing Data Upgrade when it fails and then once again run the upgrade and it's done? I guess this is the workaround?
